pkgs/by-name/cy/cy/package.nix

```diff
@@ -0,0 +1,48 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  nix-update-script,
+  go,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "cy";
+  version = "1.0.0";
+
+  src = fetchFromGitHub {
+    owner = "cfoust";
+    repo = "cy";
+    rev = "refs/tags/v${finalAttrs.version}";
+    hash = "sha256-W7xCAeoBn6nrSzkBX59qulIcgigeHhaaa38vfEEuaAQ=";
+  };
+
+  buildInputs = [ go ];
+
+  buildPhase = ''
+    export GOCACHE=$PWD/.cache/go-build
+    export GOPATH=$PWD
+    go install ./cmd/cy/
+  '';
+
+  installPhase = ''
+    install -Dm755 -t $out/bin $GOPATH/bin/cy
+  '';
```

Comment:
Add `runHook preBuild` `runHook postBuild` to buildPhase, and similar for installPhase.

Is there a reason the package does not use `buildGoModule`?

Reply:
Just rebased, added the hooks and switched to buildGoModule